extension/goext/auth.go

```diff
@@ -23,4 +25,12 @@ type IAuth interface {
 	GetTenantName(context Context) string
 	// IsAdmin reports whether context belongs to admin
 	IsAdmin(context Context) bool
+	// IsTenantIDValid checks whether given tenant_id exists in backend.
+	// Returns goext.NotValidTenantIDErr if not valid or other unspecified error
+	// and nil when provided id is valid.
+	IsTenantIDValid(ctx Context, id string) error
```

Comment:
why `error` not `bool`

Reply:
Maybe `ValidateTenantID` would be a better name.

Reply:
It should be either `error` or `(bool, error)`. 
In first approach, package must expose special value to distinguish whether tenant id is invalid or whether something unusual happened. 
In second apprach, `bool` returned distinguish whether provided value is valid or not, which in my opinion better express logic.
I'll change it to second apprach.